Dynatrace/dynatrace-operator
# Fixes OneAgent daemonset not having a separate service account for privileged mode

# Description

The OneAgent daemonset had always the same service account for both modes of operation, privileged and non-privileged.
The privileged mode needed a separate security context constraint (SCC), linked to a separate service account (SA).
This PR adds the appropriate SA, SCC and the selection of the SA when creating the daemonset.

N.B.: 
This PR also includes nil checks for certain member variables as part of some minor refactoring to ease unit testing by removing the need of having to initialize the affected variables.

## How can this be tested?

* Create an Openshift cluster, either cloud-based or with `crc`
* Deploy the operator using `make deploy/openshift` and the appropriate secret
* Apply the Dynakube
  1. normally
  2. with read-only disabled
  3. privileged
  4. with read-only disabled and privileged
* In each deployment, the OneAgent pods of the daemonset should be deployed and the security contexts of the container must match the ones defined in code.


## Checklist
